src/components/swap/PendingModalContent/Step.tsx

```diff
@@ -0,0 +1,193 @@
+
+export interface StepDetails {
+  icon: ReactElement
+  rippleColor?: string
+  previewTitle: string
+  actionRequiredTitle: string | ReactElement
+  inProgressTitle?: string
+  delayedTitle?: string
+  delayedStartTitle?: string
+  delayedEndTitle?: string
+  timerValueInSeconds?: number
+  timeToStart?: number
+  timeToEnd?: number
+  learnMoreLinkText?: string
+  learnMoreLinkHref?: string
+}
+
+export enum StepStatus {
+  PREVIEW,
+  ACTIVE,
+  IN_PROGRESS,
+  COMPLETE,
+}
+
+const outerRingAnimation = keyframes`
+  0% {
+    transform: scale(1.4);
+    opacity: 0.25;
+  }
+  50% {
+    transform: scale(1.4);
+    opacity: 0.75;
+  }
+  100% {
+    transform: scale(1);
+    opacity: 0;
+  }
+`
+const innerRingAnimation = keyframes`
+  0% {
+    transform: scale(1);
+    opacity: 0.25;
+  }
+  30% {
+    transform: scale(1.3);
+    opacity: 1;
+  }
+  60% {
+    transform: scale(1.3);
+    opacity: 0.4;
+  }
+  100% {
+    transform: scale(1);
+    opacity: 0.3;
+  }
+`
+const Ring = styled.div<{ width: string; color: string; animation: Keyframes }>`
+  position: absolute;
+  width: 24px;
+  height: 24px;
+  border: ${({ width }) => width} solid ${({ color }) => color};
+  border-radius: 50%;
+  animation: ${({ animation }) => animation} 2s linear infinite;
+`
+const IconWrapper = styled.div<{ isActive: boolean }>`
+  width: 24px;
+  height: 24px;
+  filter: ${({ isActive }) => `grayscale(${isActive ? 0 : 1})`};
+  opacity: ${({ isActive }) => (isActive ? '1' : '0.5')};
+`
+function RippleAnimation({ rippleColor }: { rippleColor?: string }) {
+  if (!rippleColor) {
+    return null
+  }
+  return (
+    <div data-testid="icon-ripple-animation">
+      <Ring width="1px" color={rippleColor} animation={innerRingAnimation} />
+      <Ring width="0.5px" color={rippleColor} animation={outerRingAnimation} />
+    </div>
+  )
+}
+
+function Icon({ stepStatus, icon, rippleColor }: { stepStatus: StepStatus; icon: ReactElement; rippleColor?: string }) {
+  if (stepStatus === StepStatus.IN_PROGRESS) {
+    return <LoaderV3 size="24px" fill={rippleColor} data-testid="loader-icon" />
+  }
+  return (
+    <div>
+      {stepStatus === StepStatus.ACTIVE && <RippleAnimation rippleColor={rippleColor} />}
+      <IconWrapper isActive={stepStatus === StepStatus.ACTIVE} data-testid="step-icon">
+        {icon}
+      </IconWrapper>
+    </div>
+  )
+}
+
+function Title({
+  stepStatus,
+  stepDetails,
+  isTimeRemaining,
+}: {
+  stepStatus: StepStatus
+  stepDetails: StepDetails
+  isTimeRemaining: boolean
+}) {
+  switch (stepStatus) {
+    case StepStatus.PREVIEW:
+      return <ThemedText.BodySecondary>{stepDetails.previewTitle}</ThemedText.BodySecondary>
+    case StepStatus.ACTIVE:
+      return isTimeRemaining ? (
+        <ThemedText.BodyPrimary>{stepDetails.actionRequiredTitle}</ThemedText.BodyPrimary>
+      ) : (
+        <ThemedText.BodyPrimary>{stepDetails.delayedStartTitle}</ThemedText.BodyPrimary>
+      )
```

Comment:
the component never changes, you can just change the content, same with the in_progress step
```suggestion
      return (
        <ThemedText.BodyPrimary>
          {isTimeRemaining ? stepDetails.actionRequiredTitle : stepDetails.delayedStartTitle}
        </ThemedText.BodyPrimary>
      )
```